### PR TITLE
Support for DeltaTable.ForName (Delta Lake 0.7+)

### PR DESCRIPTION
**Pull request by AFFogarty**
This small PR adds support for `DeltaTable.ForName`.  This API is available in Delta Lake 0.7+.

### CODE REVIEW

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs [hunk 1]
```diff
@@ -326,6 +327,22 @@ public void TestSignatures()
                 })));
         }
 
+        /// <summary>
+        /// Test that Delta Lake 0.7+ methods return the expected signature.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V3_0_0)]
+        public void TestSpark3Signatures()
```
Comment (suhsteve) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs:334]: Should we use the naming convention used in the Microsoft.Spark tests.  The one above this we should name `public void TestSignaturesV2_3_X()` maybe ?

```suggestion
        public void TestSignaturesV3_0_X()
```

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Tables/DeltaTable.cs [hunk 2]
```diff
@@ -159,6 +159,39 @@ internal DeltaTable(JvmObjectReference jvmObject)
                     sparkSession,
                     path));
 
+        /// <summary>
+        /// Create a DeltaTable using the given table or view name using the given SparkSession.
+        ///
+        /// Note: This uses the active SparkSession in the current thread to read the table data. Hence,
```
Comment (suhsteve) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Tables/DeltaTable.cs:165]: line too long.

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs (hunk 1 above)
Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs]: Fixed.

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Tables/DeltaTable.cs (hunk 2 above)
Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Tables/DeltaTable.cs]: Fixed.

**Review by suhsteve — DISMISSED** [review]
LGTM.  Thanks @AFFogarty !

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs [hunk 3]
```diff
@@ -326,6 +327,22 @@ public void TestSignatures()
...
         }
 
+        /// <summary>
+        /// Test that Delta Lake 0.7+ methods return the expected signature.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V3_0_0)]
+        public void TestSignaturesV3_0_X()
+        {
+            using var tempDirectory = new TemporaryDirectory();
+            string path = Path.Combine(tempDirectory.Path, "delta-table");
```
Comment (imback82) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs:337]: is this used?

Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs]: Ooops, removed.

**Review by imback82 — APPROVED** [review]
LGTM, thanks @AFFogarty!

**Pull request merged**